tests/test_all_rules.py

```diff
@@ -987,22 +987,31 @@ def test_event_override(self):
         errors = []
 
         for rule in self.all_rules:
-            # skip rules that do not leverage queries (i.e. machine learning)
-            # filters to acceptable query languages in definitions.FilterLanguages
-            # QueryRuleData should inherently ignore machine learning rules
-            if isinstance(rule.contents.data, QueryRuleData):
-                rule_language = rule.contents.data.language
-                has_event_ingested = rule.contents.data.get('timestamp_override') == 'event.ingested'
-                rule_str = self.rule_str(rule, trailer=None)
-
-                if not has_event_ingested:
-                    # TODO: determine if we expand this to ES|QL
-                    # ignores any rule that does not use EQL or KQL queries specifically
-                    # this does not avoid rule types where variants of KQL are used (e.g. new terms)
-                    if rule_language not in ('eql', 'kuery') or getattr(rule.contents.data, 'is_sequence', False):
-                        continue
-                    else:
-                        errors.append(f'{rule_str} - rule must have `timestamp_override: event.ingested`')
+            # Skip rules that do not leverage queries (i.e., machine learning)
+            # Exception for machine learning analytic rules
+            if not isinstance(rule.contents.data, (QueryRuleData, MachineLearningRuleData)):
+                continue
```

Comment:
all rule types inherit from both of these, so this is not needed anymore

```suggestion
```

Reply:
pending feedback of our discussion, may need to update this to _only_ skip `machine_learning` rule types

Reply:
I've asked to clarify by SecML if we intend to add `event.ingested` to all ML analytic packages or `Beaconing` only. If so, we can remove the `MachineLearningRuleData` addition. We can also update this to take other `...RuleData` schema's into consideration so it does not ignore threshold, new terms, etc.